File: examples/example-jpa/src/main/java/example/Application.java
package example;

import java.util.Arrays;

import javax.inject.Singleton;

import io.micronaut.core.annotation.TypeHint;

import org.hibernate.dialect.PostgreSQL10Dialect;
import org.postgresql.Driver;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import example.domain.Owner;
import example.domain.Pet;
import example.domain.Pet.PetType;
import example.repositories.OwnerRepository;
import example.repositories.PetRepository;
import io.micronaut.context.event.StartupEvent;
import io.micronaut.runtime.Micronaut;
import io.micronaut.runtime.event.annotation.EventListener;

@Singleton
@TypeHint( {
    PetType.class,
    PostgreSQL10Dialect.class,
    Driver.class
})
public class Application {
    private static final Logger LOG = LoggerFactory.getLogger(Application.class);

    private final OwnerRepository ownerRepository;
    private final PetRepository petRepository;

    Application(OwnerRepository ownerRepository, PetRepository petRepository) {
        this.ownerRepository = ownerRepository;
        this.petRepository = petRepository;
    }

    public static void main(String[] args) {
        Micronaut.run(Application.class);
    }

    @EventListener
    void init(StartupEvent event) {
        if (LOG.isInfoEnabled()) {
            LOG.info("Populating data");
        }

        Owner fred = new Owner();
        fred.setName("Fred");
        fred.setAge(45);
        Owner barney = new Owner();
        barney.setName("Barney");
        barney.setAge(40);
        ownerRepository.saveAll(Arrays.asList(fred, barney));

        Pet dino = new Pet();
        dino.setName("Dino");
        dino.setOwner(fred);
        Pet bp = new Pet();
        bp.setName("Baby Puss");
        bp.setOwner(fred);
        bp.setType(PetType.CAT);
        Pet hoppy = new Pet();
        hoppy.setName("Hoppy");
        hoppy.setOwner(barney);

        petRepository.saveAll(Arrays.asList(dino, bp, hoppy));
    }
}

File: examples/example-jdbc-kotlin/src/main/kotlin/example/Application.kt
package example

import example.domain.Owner
import example.domain.Pet
import example.domain.Pet.PetType
import example.repositories.OwnerRepository
import example.repositories.PetRepository
import io.micronaut.context.event.StartupEvent
import io.micronaut.core.annotation.TypeHint
import io.micronaut.runtime.Micronaut
import io.micronaut.runtime.event.annotation.EventListener
import org.slf4j.LoggerFactory
import javax.inject.Singleton

fun main(args: Array<String>) {
    Micronaut.run(*args)
}

@Singleton
@TypeHint(typeNames = ["org.h2.Driver", "org.h2.mvstore.db.MVTableEngine"])
class Application(
        private val ownerRepository: OwnerRepository,
        private val petRepository: PetRepository) {

    @EventListener
    fun init(event: StartupEvent) {
        if (LOG.isInfoEnabled) {
            LOG.info("Populating data")
        }

        val fred = Owner(0, "Fred", 45)
        val barney = Owner(0, "Barney",40)
        ownerRepository.saveAll(listOf(fred, barney))

        val dino = Pet(null, "Dino", fred)
        val bp = Pet(null, "Baby Puss", fred, PetType.CAT)
        val hoppy = Pet(null, "Hoppy", barney)

        petRepository.saveAll(listOf(dino, bp, hoppy))
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(Application::class.java)
    }
}